packages/itwin/grouping-mapping-widget/src/components/Mappings/hooks/useFetchExtractionStates.ts

```diff
@@ -0,0 +1,76 @@
+/*---------------------------------------------------------------------------------------------
+* Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+* See LICENSE.md in the project root for license terms and full copyright notice.
+*--------------------------------------------------------------------------------------------*/
+import { useExtractionClient } from "../../context/ExtractionClientContext";
+import { ExtractorState } from "@itwin/insights-client";
+import { useMutation, useQuery } from "@tanstack/react-query";
+import type { GroupingMappingApiConfig } from "../../context/GroupingApiConfigContext";
+import type { ExtractionRequestMapping, ExtractionRunRequest, Mapping } from "@itwin/insights-client";
+import { useState } from "react";
+import { ExtractionStates } from "../Extraction/ExtractionStatus";
+import { STATUS_CHECK_INTERVAL } from "../../Constants";
+
+export const useFetchExtractionStates = ({
+  iModelId,
+  getAccessToken,
+}: GroupingMappingApiConfig) => {
+
+  const extractionClient = useExtractionClient();
+  const [isJobStarted, setIsJobStarted] = useState<boolean>(false);
+  const [isJobDone, setIsJobDone] = useState<boolean>(false);
+
+  const { mutateAsync: runExtraction } = useMutation({
+    mutationKey: ["runExtraction"],
+    mutationFn: async (mappings: Mapping[]) => {
+      const accessToken = await getAccessToken();
+      const mappingIds: ExtractionRequestMapping[] = mappings.length > 0 ? mappings.map((mapping) => { return { id: mapping.id }; }) : [];
+      const extractionRequest: ExtractionRunRequest | undefined = {
+        mappings: mappingIds,
+      };
+      await extractionClient.runExtraction(accessToken, iModelId, extractionRequest);
+    },
+  });
+
+  const { data: extractionStateData, isError: isExtractionStateError, isLoading: isExtractionStatusLoading } = useQuery({
+    queryKey: ["extractionState"],
+    staleTime: Infinity,
+    placeholderData: undefined,
+    queryFn: async () => {
+      const accessToken = await getAccessToken();
+      const extractionHistoryIterator = extractionClient.getExtractionHistoryIterator(accessToken, iModelId, 1);
+      const extractionHistoryResult = await extractionHistoryIterator.next();
+      const jobId = extractionHistoryResult.value.jobId;
```

Comment:
This query will fail if used on an iModel with no extractions.

Reply:
I have made it to check whether `jobId` exists, if not just simply returns the "None" status. I'm not sure if this works effectively.